packages/artifact/src/internal/download-http-client.ts

```diff
@@ -112,10 +138,54 @@ export class DownloadHttpClient {
     artifactLocation: string,
     downloadPath: string
   ): Promise<void> {
-    const stream = fs.createWriteStream(downloadPath)
-    const client = this.downloadHttpManager.getClient(httpClientIndex)
-    const requestOptions = getRequestOptions('application/octet-stream', true)
-    const response = await client.get(artifactLocation, requestOptions)
+    let retryCount = 0
+    const retryLimit = getRetryLimit()
+    const destinationStream = fs.createWriteStream(downloadPath)
+    const requestOptions = getDownloadRequestOptions(
+      'application/json',
+      true,
+      true
+    )
+
+    // a single GET request is used to download a file
+    const makeDownloadRequest = async (): Promise<IHttpClientResponse> => {
+      const client = this.downloadHttpManager.getClient(httpClientIndex)
+      return await client.get(artifactLocation, requestOptions)
+    }
+
+    // Increments the current retry count and then checks if the retry limit has been reached
+    // If there have been too many retries, fail so the download stops
+    const incrementAndCheckRetryLimit = (
+      response?: IHttpClientResponse
+    ): void => {
+      retryCount++
+      if (retryCount > retryLimit) {
+        if (response) {
+          // display extra information if the retry limit has been reached
+          // eslint-disable-next-line no-console
+          console.log(response)
+        }
+        throw new Error(
+          `Unable to download ${artifactLocation}. Retry limit has been reached`
+        )
+      }
+    }
+
+    // Back off exponentially based off of the retry count
+    const backoffExponentially = async (): Promise<void> => {
```

Comment:
I would suggest DRY here again.  This function is largely the same as the other `backOffUsingRetryValue`, both can be simplified to a single function `backoff(timespan)`.

Reply:
Also, if `incrementAndCheckRetryLimit` is **always** called before backoff exponentially, you can even fold that method into `backoff`.  If a value is passed it, use that timeout value; otherwise calculate the retry automatically and backoff.

Reply:
I did some pretty big refactoring in both` upload-http-client` and `download-http-client`

In combined both of the back off functions into a single `backOff()` for both http clients, while in download I also combined `incrementAndCheckRetryLimit()` into the same function. Simplifies it a little bit more.